samples/wifi/radio_test/src/nrf_wifi_radio_test_shell.c

```diff
@@ -289,6 +289,51 @@ enum wifi_nrf_status nrf_wifi_radio_test_conf_init(struct rpu_conf_params *conf_
 {
 	enum wifi_nrf_status status = WIFI_NRF_STATUS_FAIL;
 	unsigned char country_code[NRF_WIFI_COUNTRY_CODE_LEN] = {0};
+	struct nrf_wifi_tx_pwr_ceil_params tx_pwr_ceil_params;
+
+#if defined(CONFIG_BOARD_NRF7002DK_NRF7001_NRF5340_CPUAPP) || \
+defined(CONFIG_BOARD_NRF7002DK_NRF5340_CPUAPP) || \
+defined(CONFIG_BOARD_NRF5340DK_NRF5340_CPUAPP)
+	memset(&tx_pwr_ceil_params, 0, sizeof(tx_pwr_ceil_params));
+	#if DT_NODE_EXISTS(DT_NODELABEL(nrf70_tx_power_ceiling))
+		tx_pwr_ceil_params.max_pwr_2g_dsss =
+				DT_PROP(DT_NODELABEL(nrf70_tx_power_ceiling), max_pwr_2g_dsss);
+		tx_pwr_ceil_params.max_pwr_2g_mcs7 =
+				DT_PROP(DT_NODELABEL(nrf70_tx_power_ceiling), max_pwr_2g_mcs7);
+		tx_pwr_ceil_params.max_pwr_2g_mcs0 =
+				DT_PROP(DT_NODELABEL(nrf70_tx_power_ceiling), max_pwr_2g_mcs0);
+		tx_pwr_ceil_params.rf_tx_pwr_ceil_params_override = 1;
+	#else
+		tx_pwr_ceil_params.max_pwr_2g_dsss = 84;
+		tx_pwr_ceil_params.max_pwr_2g_mcs7 = 64;
+		tx_pwr_ceil_params.max_pwr_2g_mcs0 = 64;
+		tx_pwr_ceil_params.rf_tx_pwr_ceil_params_override = 0;
+	#endif
+#endif
+
+#if defined(CONFIG_BOARD_NRF7002DK_NRF5340_CPUAPP) || defined(CONFIG_BOARD_NRF5340DK_NRF5340_CPUAPP)
+	#if DT_NODE_EXISTS(DT_NODELABEL(nrf70_tx_power_ceiling))
+		tx_pwr_ceil_params.max_pwr_5g_low_mcs7 =
+			DT_PROP(DT_NODELABEL(nrf70_tx_power_ceiling), max_pwr_5g_low_mcs7);
+		tx_pwr_ceil_params.max_pwr_5g_mid_mcs7 =
+			DT_PROP(DT_NODELABEL(nrf70_tx_power_ceiling), max_pwr_5g_mid_mcs7);
+		tx_pwr_ceil_params.max_pwr_5g_high_mcs7 =
+			DT_PROP(DT_NODELABEL(nrf70_tx_power_ceiling), max_pwr_5g_high_mcs7);
+		tx_pwr_ceil_params.max_pwr_5g_low_mcs0 =
+			DT_PROP(DT_NODELABEL(nrf70_tx_power_ceiling), max_pwr_5g_low_mcs0);
+		tx_pwr_ceil_params.max_pwr_5g_mid_mcs0 =
+			DT_PROP(DT_NODELABEL(nrf70_tx_power_ceiling), max_pwr_5g_mid_mcs0);
+		tx_pwr_ceil_params.max_pwr_5g_high_mcs0 =
+			DT_PROP(DT_NODELABEL(nrf70_tx_power_ceiling), max_pwr_5g_high_mcs0);
+	#else
+		tx_pwr_ceil_params.max_pwr_5g_low_mcs7 = 56;
```

Comment:
Same as above, has unconditional defaults and then only the overrides.

Reply:
All moved to Init function.